### DISCUSSION TITLE
Embedding of Firefox

### DISCUSSION THREAD

**Discussion by madsl** [filips123/PWAsForFirefox#409] [Q&A] [answered]
A quick one! It seems like this project actually builds and embeds its own firefox build? Is this right?

Or what actually runs when you run "firefoxpwa"? It seemed like the program still works even if I remove firefox from the computer...

**Comment by filips123** [comment, marked as answer]
During the initial setup wizard (or when you run `firefoxpwa runtime install`), the native program will download a separate Firefox version directly from Mozilla and store it to [a user-specific directory](https://pwasforfirefox.filips.si/resources/installation-directories/#profiles). This is needed as it's hard to re-use an existing Firefox installation due to technical reasons.

Alternatively, you can also manually copy Firefox to the correct directory, do some tricks with symlinks, or [link it with FUSE OverlayFS](https://pwasforfirefox.filips.si/help/faq/#how-to-use-fuse-overlayfs-with-the-app-browser).

**Comment by madsl** [comment]
Thanks for the reply, that was illuminating!
